src/sentry/seer/anomaly_detection/utils.py

```diff
@@ -143,37 +173,52 @@ def fetch_historical_data(
     granularity = snuba_query.time_window
 
     dataset_label = snuba_query.dataset
+
     if dataset_label == "events":
         # DATASET_OPTIONS expects the name 'errors'
         dataset_label = "errors"
     elif dataset_label == "generic_metrics":
+        # XXX: performance alerts in prod
         dataset_label = "transactions"
```

Comment:
not sure if I need to be using the discover dataset here, it's hard to know since it's different locally

Reply:
I think we _do_ want to use the discover dataset - unfortunately this differs in prod vs. locally so it's confusing but locally we make this request:

`api/0/organizations/sentry/events-stats/?interval=60m&project=1&query=event.type%3Atransaction&referrer=api.organization-event-stats&statsPeriod=9998m&yAxis=count%28%29`

and in prod we make this request:

`api/0/organizations/sentry/events-stats/?dataset=metricsEnhanced&interval=60m&project=1&query=event.type%3Atransaction&referrer=api.organization-event-stats&statsPeriod=9998m&yAxis=count%28%29`

the `events-stats` endpoint gets the dataset label from the request [here](https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization_events.py#L83-L89) and falls back to using discover if nothing is passed (like in production). 

In our case we're not getting the dataset label from the request but rather the snuba query object, so instead of it being "discover" it's "generic_metrics" in production and "transactions" in my local db. 

All that to say for performance metrics alerts we need to use the discover dataset for both `generic_metrics` and `transaction` dataset labels